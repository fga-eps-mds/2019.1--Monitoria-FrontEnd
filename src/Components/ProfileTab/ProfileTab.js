import React from 'react';
import Paper from '@material-ui/core/Paper';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';


const theme = createMuiTheme({
  palette: {
    primary: { main: '#44a1f2' },
    secondary: { main: '#ff0000' },
  },
  typography: { useNextVariants: true },
  overrides: {
      MuiButton: {
        raisedPrimary: {
          color: 'white',
        },
      },
  },
});
class ProfileTab extends React.Component {
  state = {
    value: 0,
  };

  handleChange = (event, value) => {
    this.setState({ value });
    
    
  };

  render() {
    return (
      <Paper square >
      <MuiThemeProvider theme={theme}>
        <Tabs
          value={this.state.value}
          indicatorColor="primary"
          textColor="primary"
          onChange={this.handleChange}
          variant="fullWidth"
          centered
        >
          <Tab value="1" label="A ministrar" />
          <Tab value="2" label="Curtidas"  />
        </Tabs>
        </MuiThemeProvider>
      </Paper>
    );
  }
}

export default ProfileTab;