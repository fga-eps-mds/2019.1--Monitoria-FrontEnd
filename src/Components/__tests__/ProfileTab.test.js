import React from 'react';
import ProfileTab from '../ProfileTab/ProfileTab';
import { shallow , mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Testing ProfileTab component', () => {
    it('Test if ProfileTab renders correctly', () =>{
        const tree = shallow(<ProfileTab/>);
        expect(toJson(tree)).toMatchSnapshot();
    });
    it('allows us to set props', () => {
        const wrapper = mount(<ProfileTab bar="baz" />);
        expect(wrapper.props().bar).toMatchSnapshot('baz');
        wrapper.setProps({ bar: 'profiletab' });
        expect(wrapper.props().bar).toMatchSnapshot('profiletab');
    });
});