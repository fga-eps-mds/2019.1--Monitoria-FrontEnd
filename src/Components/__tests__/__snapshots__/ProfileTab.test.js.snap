// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProfileTab component Test if ProfileTab renders correctly 1`] = `
<WithStyles(Paper)
  square={true}
>
  <WithStyles(Tabs)
    centered={true}
    indicatorColor="primary"
    onChange={[Function]}
    textColor="primary"
    value={0}
    variant="fullWidth"
  >
    <WithStyles(Tab)
      label="A ministrar"
    />
    <WithStyles(Tab)
      disabled={true}
      label="A assistir"
    />
    <WithStyles(Tab)
      disabled={true}
      label="Historico"
    />
  </WithStyles(Tabs)>
</WithStyles(Paper)>
`;

exports[`Testing ProfileTab component allows us to set props: baz 1`] = `"baz"`;

exports[`Testing ProfileTab component allows us to set props: profiletab 1`] = `"profiletab"`;
